changelog/unreleased/issue-3267

```diff
@@ -0,0 +1,7 @@
+Bugfix: `copy` failed to copy snapshots in rare cases
+
+The `copy` could in rare cases fail with the error message `SaveTree(...)
+returned unexpected id ...`.  This has been fixed.
```

Comment:
I would mention symlinks with non-UTF-8 targets here, and also explain that they're still somewhat problematic due to limitations of restic's repo format and that users may want to avoid them if possible.

Reply:
I wonder whether that is the best place to add that information. After all the failing copy command is just the symptom and not the cause.

Reply:
Perhaps a warning in the documentation for the copy command?

I'm thinking about the type of notice that's shown in https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html#local or in https://restic.readthedocs.io/en/stable/045_working_with_repos.html#copying-snapshots-between-repositories depending on the severity.

Reply:
The warning would rather belong into the documentation of the `backup` command. That is the point at which invalid UTF8 characters in symlinks are replaced. Once this PR is merged, the copy command won't care at all about the symlinks.

Reply:
My rationale for adding it here is that this is what users are going to find when they search for the error message. I'm not saying it shouldn't be addressed elsewhere too :)

Reply:
I've amended the changelog